src/signalrclient/connection_impl.cpp

```diff
@@ -397,53 +424,6 @@ namespace signalr
                 }
             });
 
-        disconnect_cts->register_callback([connect_request_done, connect_request_lock, callback]()
```

Comment:
I haven't looked at this super carefully yet, but is it not a problem that there are two declarations of `std::shared_ptr<bool> connect_request_done = std::make_shared<bool>();` now? Do these represent different things now? If so, can we use slightly more descriptive names??

Reply:
Nice catch, I believe the old one can be removed